OpenRA.Mods.Common/Traits/Armament.cs

```diff
@@ -62,6 +62,19 @@ public class ArmamentInfo : UpgradableTraitInfo, Requires<AttackBaseInfo>
 		[Desc("Use multiple muzzle images if non-zero")]
 		public readonly int MuzzleSplitFacings = 0;
 
+		[Desc("Is it possible to use the armament when force-attacking?")]
+		public readonly bool ForceAttacks = true;
+
+		[Desc("Is the armament beneficial (should it be used on allies)?")]
+		public readonly bool Beneficial = false;
```

Comment:
Force-attack ignores this tag. Keep in mind this is used for target acquisition to define it. Also, `CanTargetAllies: false` wouldn't suggest that the weapon is used on enemies.